File: memo/src/main/java/com/zeoflow/memo/ZsonParser.java
package com.zeoflow.memo;

import android.text.TextUtils;

import com.zeoflow.zson.JsonSyntaxException;
import com.zeoflow.zson.Zson;

import java.lang.reflect.Type;

public final class ZsonParser implements Parser
{

    private final Zson zson;

    public ZsonParser(Zson zson)
    {
        this.zson = zson;
    }

    @Override
    public <T> T fromJson(String content, Type type) throws JsonSyntaxException
    {
        if (TextUtils.isEmpty(content))
        {
            return null;
        }
        return zson.fromJson(content, type);
    }

    @Override
    public String toJson(Object body)
    {
        return zson.toJson(body);
    }

}

File: memo/src/main/java/com/zeoflow/memo/ZsonParser.kt
package com.zeoflow.memo

import android.text.TextUtils
import com.zeoflow.zson.JsonSyntaxException
import com.zeoflow.zson.Zson
import java.lang.reflect.Type

class ZsonParser(private val zson: Zson) : IParser {
    @Throws(JsonSyntaxException::class)
    override fun <T> fromJson(content: String?, type: Type?): T? {
        return if (TextUtils.isEmpty(content)) {
            null
        } else zson.fromJson(content, type)
    }

    override fun toJson(body: Any?): String? {
        return zson.toJson(body)
    }
}